haystack/dataclasses/streaming_chunk.py

```diff
@@ -102,6 +102,71 @@ def __post_init__(self):
         if (self.tool_calls or self.tool_call_result) and self.index is None:
             raise ValueError("If `tool_call`, or `tool_call_result` is set, `index` must also be set.")
 
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary representation of the StreamingChunk.
+
+        :returns: Serialized dictionary representation of the calling object.
+        """
+        return {
+            "content": self.content,
+            "meta": self.meta,
+            "component_info": asdict(self.component_info) if self.component_info else None,
+            "index": self.index,
+            "tool_calls": [asdict(tc) for tc in self.tool_calls] if self.tool_calls else None,
+            "tool_call_result": asdict(self.tool_call_result) if self.tool_call_result else None,
```

Comment:
Same [comment](https://github.com/deepset-ai/haystack/pull/9608/files#r2218294498) for these two fields here. Let's create `to_dict` and `from_dict` methods for each dataclass.